frontend/packages/console-dynamic-plugin-sdk/src/app/kinds.ts

```diff
@@ -0,0 +1,17 @@
+import * as _ from 'lodash';
+import { connect } from 'react-redux';
+import { K8sKind, K8sResourceKindReference, kindForReference } from '@console/internal/module/k8s'; // ??! TODO - handle them
+import { SDKStoreState } from './redux-types';
+
+export const connectToModel = connect(
+  (state: SDKStoreState, props: { kind: K8sResourceKindReference } & any) => {
+    const kind: string = props.kind || _.get(props, 'match.params.plural');
+
+    const kindObj: K8sKind = !_.isEmpty(kind)
+      ? state.k8s.getIn(['RESOURCES', 'models', kind]) ||
```

Comment:
@fhlavac having a nested conditional inside of a ternary makes my head hurt... Can we write this to be a bit more readable? I've also typically tried avoiding having `null` as the default value in ternaries.